File: engine/model/src/main/scala/tools/packet/monster/KillMonster.scala
package tools.packet.monster

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class KillMonster(private var _objectId: Int, private var _animation: Int) extends PacketInput {
  def objectId: Int = _objectId

  def animation: Int = _animation

  def this(_objectId: Int, _animation: Boolean) = {
    this(_objectId, if (_animation) 1 else 0)
  }

  override def opcode(): SendOpcode = SendOpcode.KILL_MONSTER
}

File: engine/engine-model/src/main/java/tools/packet/monster/KillMonster.java
package tools.packet.monster;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record KillMonster(Integer objectId, Integer animation) implements PacketInput {
   public KillMonster(Integer objectId, Boolean animation) {
      this(objectId, animation ? 1 : 0);
   }

   @Override
   public SendOpcode opcode() {
      return SendOpcode.KILL_MONSTER;
   }
}